src/pybind/mgr/dashboard_v2/HACKING.rst

```diff
@@ -508,3 +508,160 @@ The settings management implementation will make sure that if you change a
+
+The ``TaskManager.run`` method triggers the asynchronous execution of function
+``func`` and returns an ``AsyncTask`` object.
+The ``AsyncTask`` provides the public method ``AsyncTask.wait(timeout)``, which
+can be used to wait for the task to complete up to a timeout defined in seconds
+and provided as an argument. If not argument is provided the ``wait`` method
+blocks until the task is finished.
+
+The ``AsyncTask.wait`` is very useful for tasks that usually are fast
+to execute but that sometimes may take a long time to run.
+The return value of the ``AsyncTask.wait`` method is a pair ``(state, value)``
+where ``state`` is an integer with following possible values:
+
+* ``VALUE_DONE = 0``
+* ``VALUE_EXECUTING = 1``
+* ``VALUE_EXCEPTION = 2``
+
+The ``value`` will store the result of the execution of function ``func`` if
+``state == VALUE_DONE``. If ``state == VALUE_EXECUTING`` then
+``value == None``, and if ``state == VALUE_EXCEPTION`` then ``value`` stores
+the exception object raised by the execution of function ``func``.
+
+The pair ``(namespace, metadata)`` should univocally identify the task being
+run, which means that if you try to trigger a new task that matches the same
+``(namespace, metadata)`` pair of the currently running task, then the new task
+is not created and you get the task object of the current running task.
+
+
+How to get the list of executing and finished asynchronous tasks?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The list of executing and finished tasks is included in the ``Summary``
+controller, which is already polled every 5 seconds by the dashboard frontend.
+But we also provide a dedicated controller to get the same list of executing
+and finished tasks.
+
+The ``Task`` controller exposes the ``/api/task`` endpoint that returns the
+list of executing and finished tasks. This endpoint accepts the ``namespace``
+parameter that accepts a glob expression as its value.
+For instance, an HTTP GET request of the URL ``/api/task?namespace=rbd/*``
+will return all executing and finished tasks which namespace starts with
+``rbd/``.
+
+To prevent the finished tasks list from growing unboundly, the finished tasks
+will be maintained in memory for 1 minute. After a minute, when the finished
+task information is retrieved, either by the summary controller or by the task
+controller, it is automatically deleted from the list and it will not be
+included in further task queries.
+
+Each executing task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'progress': 0  # int (percentage)
+  }
+
+Each finished task is represented by the following dictionary::
+
+  {
+    'namespace': "namespace",  # str
+    'metadata': { },  # dict
+    'begin_time': 0.0,  # float
+    'end_time': 0.0,  # float
+    'latency': 0.0,  # float
+    'progress': 0  # int (percentage)
+    'success': True,  # bool
```

Comment:
I'd replace with `state` add unify executing and finished tasks. This would also improve the TaskManager itself.